tools/status_check.py

```diff
@@ -0,0 +1,138 @@
+# Validate that the framecounts in STATUS.md are up-to-date. This is done by trying to run the entire test case and
+# asserting that the sync framecount is the highest it currently can be. This helps us capture when changes focused
+# on one test case actually benefit a different test case.
+
+from generate_tests import generate_tests
+import json
+import os
+import subprocess
+import sys
+from typing import Dict
+
+STATUS_TEST_CASE_FILENAME = 'statusTestCases.json'
+
+
+class TestCase:
+    def __init__(self, line: str):
+        # Get the test case name in between the square brackets
+        self.name = line[line.find('[') + 2:line.find(']') - 1]
+
+        # Get the frame counts in between the pipes
+        frame_counts = line.split('|')[1].split('/')
```

Comment:
What are your thoughts on isolating these pieces of line logic into their own helper functions within the class?

Also, what are your thoughts on validating the integrity of the line before operating on it?

It may also be worth storing the line in the class as a result of this.

Reply:
With a script like this I personally would only make those helper functions if it helped readability. I think the logic (along with the comments explaining) is straight-forward enough that they don't need helper functions. The stdout parsing is a lot more convoluted which is why I definitely vouch for helper functions there.

Regarding line validation, I'm lazy. My thought is that any malformed input will cause the program to crash, which will fail the build action, thus we can observe that something is wrong with STATUS.md and fix it accordingly.

TL;DR: I don't feel like it's necessary but if you feel strongly then sure on both parts